m2m/m2m.py

```diff
@@ -0,0 +1,263 @@
+import csv
+import time
+import os
+import sys
+import json
+from optparse import OptionParser
+
+from pyuntl.untl_structure import PYUNTL_DISPATCH
+from pyuntl.untldoc import untlpydict2xmlstring, untlpy2dict
+
+
+fieldTypes = {
+    "title": "basic",
+    "creator": "agent",
+    "contributor": "agent",
+    "publisher": "agent",
+    "date": "basic",
+    "language": "basic",
+    "description": "basic",
+    "subject": "basic",
+    "coverage": "basic",
+    "source": "basic",
+    "relation": "basic",
+    "collection": "basic",
+    "institution": "basic",
+    "rights": "basic",
+    "resourceType": "basic",
+    "format": "basic",
+    "identifier": "basic",
+    "note": "basic",
+    "degree": "basic",
+    "meta": "basic",
+    "primarySource": "basic",
+    "citation": "basic",
+    }
+
+
+def CSVToDict(csvFileName):
+    readerDict = csv.DictReader(open(csvFileName))
+    readerList = []
+    for d in readerDict:
+        for k, v in d.items():
+            try:
+                d[k] = unicode(v, 'utf-8', 'replace')  # probably not smart
+            except:
+                pass  # not sure how smart /this/ is either...
+        readerList.append(d)
+    return readerList
+
+
+class MetadataConverterException(Exception):
+    """This is our base exception for this script"""
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        return "%s" % (self.value,)
```

Comment:
I am not 100% sure what the scope of this PR is, but it we should change this at some point. 

We can get the same thing with
```python
class MetadataConverterException(Exception):
    pass
```

Then we are not changing where the exception message is stored or its string representation. (The message is stored in the `message` attribute by default.)